microformats.go

```diff
@@ -248,12 +276,18 @@ func (p *parser) walk(node *html.Node) {
 				}
 			case "e":
 				value = new(string)
-				*value = getTextContent(node)
+				*value = strings.TrimSpace(getTextContent(node))
 				var buf bytes.Buffer
+
 				for c := node.FirstChild; c != nil; c = c.NextSibling {
+					p.replaceHref(c)
 					html.Render(&buf, c)
 				}
-				htmlbody = strings.TrimSpace(buf.String())
+
+				htmlbody = strings.Replace(buf.String(), "/>", " />", -1)
```

Comment:
so, this is going to be pretty fragile.  If the testsuite happens to have an example that *doesn't* have this space, then those tests will fail. So instead of trying to normalize it here, we should normalize the test strings before comparison.